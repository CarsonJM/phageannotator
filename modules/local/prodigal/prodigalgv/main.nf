process PRODIGAL_PRODIGALGV {
    tag "$meta.id"
    label 'process_high'

    conda "${moduleDir}/environment.yml"
    container "${ workflow.containerEngine == 'singularity' && !task.ext.singularity_pull_docker_container ?
        'https://depot.galaxyproject.org/singularity/prodigal-gv:2.11.0--he4a0461_2':
        'biocontainers/prodigal-gv:2.11.0--he4a0461_2' }"

    input:
    tuple val(meta), path(fasta)

    output:
    tuple val(meta), path("*.prodigalgv.faa.gz")    , emit: faa
    tuple val(meta), path("*.prodigalgv.fna.gz")    , emit: fna
    path "versions.yml"                             , emit: versions

    when:
    task.ext.when == null || task.ext.when

    script:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    prodigal-gv \\
        -i $fasta \\
        -d ${prefix}.prodigalgv.fna \\
        -a ${prefix}.prodigalgv.faa \\
        $args

    gzip ${prefix}.prodigalgv.fna
    gzip ${prefix}.prodigalgv.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        prodigalgv: \$(echo \$(prodigal-gv --version 2>&1) | awk 'NF{ print \$NF }')
    END_VERSIONS
    """

    stub:
    def args = task.ext.args ?: ''
    def prefix = task.ext.prefix ?: "${meta.id}"
    """
    touch ${prefix}.prodigalgv.fna
    touch ${prefix}.prodigalgv.faa

    cat <<-END_VERSIONS > versions.yml
    "${task.process}":
        prodigalgv: \$(echo \$(prodigal-gv --version 2>&1) | awk 'NF{ print \$NF }')
    END_VERSIONS
    """
}
