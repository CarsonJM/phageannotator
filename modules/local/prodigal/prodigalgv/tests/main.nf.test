nextflow_process {

    name "Test Process: PRODIGAL_PRODIGALGV"
    script "../main.nf"
    process "PRODIGAL_PRODIGALGV"
    tag "modules"
    tag "modules_local"
    tag "prodigal"
    tag "prodigal/prodigalgv"


    test("['sarscov2']['genome']['genome_fasta']") {

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'all_samples' ], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true) ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }

    test("['sarscov2']['genome']['genome_fasta'] - stub") {

        options "-stub"

        when {
            params {
                outdir   = "$outputDir"
            }
            process {
                """
                input[0] = [ [ id: 'all_samples' ], file(params.test_data['sarscov2']['genome']['genome_fasta'], checkIfExists: true) ]
                """
            }
        }

        then {
            assertAll (
                { assert process.success },
                { assert snapshot(process.out).match() }
            )
        }
    }
}
