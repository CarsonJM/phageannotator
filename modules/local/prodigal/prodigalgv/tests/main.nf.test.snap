{
    "['sarscov2']['genome']['genome_fasta']": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,6f28a3b4d319835fe5ce4864bc3b7c2c"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.fna.gz:md5,d6eb37cf9a223db5dcf56538f5050c76"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3849079cb0a073e302ead9d90e4624d4"
                ],
                "faa": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,6f28a3b4d319835fe5ce4864bc3b7c2c"
                    ]
                ],
                "fna": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.fna.gz:md5,d6eb37cf9a223db5dcf56538f5050c76"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3849079cb0a073e302ead9d90e4624d4"
                ]
            }
        ],
        "timestamp": "2023-12-06T15:25:14.501666251"
    },
    "['sarscov2']['genome']['genome_fasta'] - stub": {
        "content": [
            {
                "0": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "1": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "2": [
                    "versions.yml:md5,3849079cb0a073e302ead9d90e4624d4"
                ],
                "faa": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.faa.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "fna": [
                    [
                        {
                            "id": "all_samples"
                        },
                        "all_samples.prodigalgv.fna.gz:md5,d41d8cd98f00b204e9800998ecf8427e"
                    ]
                ],
                "versions": [
                    "versions.yml:md5,3849079cb0a073e302ead9d90e4624d4"
                ]
            }
        ],
        "timestamp": "2024-01-17T22:52:02.155249524"
    }
}
